bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsBridgeHandler.java

```diff
@@ -0,0 +1,880 @@
+    private volatile boolean restart;
+    static final int RESTART_DELAY_MS = 1000;
+
+    private volatile String connectorId = "";
+    // Used for getting IP address and keep connection alive messages
+    private static final String QUERY_BASE_STRING = "IOT_KEY?";
+    private volatile String queryString = QUERY_BASE_STRING + connectorId;
+
+    private int refreshInterval = 60;
+    private volatile @Nullable InetAddress addr;
+    private volatile String ctrlKey = "";
+
+    private volatile @Nullable DatagramSocket socket;
+    private volatile @Nullable DatagramPacket ackPacket;
+
+    private volatile @Nullable ScheduledFuture<?> syncFuture;
+    private volatile boolean keepRunning = false;
+    private volatile @Nullable CompletableFuture<Boolean> awaitResponse;
+
+    private ElroConnectsDynamicStateDescriptionProvider stateDescriptionProvider;
+
+    private final Map<Integer, String> scenes = new ConcurrentHashMap<>();
+    private final Map<Integer, ElroConnectsDevice> devices = new ConcurrentHashMap<>();
+    private final Map<Integer, ElroConnectsDeviceHandler> deviceHandlers = new ConcurrentHashMap<>();
+
+    private int currentScene;
+
+    // We only keep 2 gson adapters used to serialize and deserialize all messages sent and received
+    private final Gson gsonOut = new Gson();
+    private Gson gsonIn = new Gson();
+
+    public ElroConnectsBridgeHandler(Bridge bridge, NetworkAddressService networkAddressService,
+            ElroConnectsDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(bridge);
+        this.networkAddressService = networkAddressService;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+
+        resetScenes();
+    }
+
+    @Override
+    public void initialize() {
+        ElroConnectsBridgeConfiguration config = getConfigAs(ElroConnectsBridgeConfiguration.class);
+
+        connectorId = config.connectorId;
+        refreshInterval = config.refreshInterval;
+        if (connectorId.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Device ID not set");
+            return;
+        }
+
+        queryString = QUERY_BASE_STRING + connectorId;
+
+        scheduler.submit(this::startCommunication);
+    }
+
+    @Override
+    public void dispose() {
+        stopCommunication();
+    }
+
+    private synchronized void startCommunication() {
+        if (this.connectorId.isEmpty()) {
+            // Should not happen, as this is set in initialize()
+            return;
+        }
+
+        InetAddress addr = null;
+        try {
+            addr = getAddr();
+        } catch (IOException e) {
+            logger.warn("Communication error when finding IP address.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error trying to find IP address for connector ID " + connectorId + ".");
```

Comment:
See https://next.openhab.org/docs/developer/guidelines.html#f-logging

Suggest you remove the logger.warn as any status update already gets logged to the event.log